### PR TITLE
feat(profiling): support protobuf 4.x

### LINKED ISSUE #3791: protobuf v4 (4.21.0) compatibility
Hi DataDog,

Would it be possible to add protobuf v4 support?

https://github.com/DataDog/dd-trace-py/blob/1.x/setup.py#L244

https://github.com/protocolbuffers/protobuf/releases/tag/v21.0

https://pypi.org/project/protobuf/

Thank you!

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
4.21 dropped support for directly creating descriptors which the
 generated profiling protobuf code was using.

Protobuf states:

> If this call came from a _pb2.py file, your generated code is out of
date and must be regenerated with protoc >= 3.19.0.

And so rename `pprof_pb2.py` to `pprof_pre319_pb2.py` and regenerate
`pprof_pb2.py` using protoc 3.19.4:

```sh
$ cd ddtrace/profiling/exporter
$ protoc --version
libprotoc 3.19.4
$ protoc --proto_path=. --python_out=. pprof.proto
```

Fixes #3791 

## Checklist
- [x] Library documentation is updated.
- [x] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
no tox changes?

**Comment by Kyle-Verhoog** [comment]
I _think_ the tox expression used by CI should pick up the added environment.. Waiting on CI to confirm this.

**Comment by Kyle-Verhoog** [comment]
```
  py37-profile-protobuf319: commands succeeded
```
https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/17738/workflows/6e52dbfc-3818-47ad-84d8-597e051c833f/jobs/1207736/parallel-runs/5?filterBy=ALL


😎

**Review by brettlangdon — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -240,8 +240,8 @@ def get_exts_for(name):
         "funcsigs>=1.0.0; python_version=='2.7'",
         "typing; python_version<'3.5'",
         "packaging>=17.1",
+        "protobuf>=3,<4.0; python_version<'3.7'",
```
Comment (brettlangdon) [inline setup.py:243]: what about Python >=3.8 ?

Comment (brettlangdon) [reply setup.py:243]: how does this play with `python_version<=3.6` ? not sure I've played with overlapping env constraints like this before

**Review by brettlangdon — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply setup.py:244]: yeah not sure either

File: setup.py [hunk 2]
```diff
@@ -240,8 +240,9 @@ def get_exts_for(name):
         "funcsigs>=1.0.0; python_version=='2.7'",
         "typing; python_version<'3.5'",
         "packaging>=17.1",
+        "protobuf>=3; python_version>='3.7'",
+        "protobuf>=3,<4.0; python_version<'3.7'",
```
Comment (P403n1x87) [inline setup.py:244]: can we take a similar approach to the `bytecode` dependencies to tidy this up a bit?

Comment (brettlangdon) [reply setup.py:244]: we can, and should do that in a follow-up PR

**Review by brettlangdon — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -414,6 +414,7 @@ jobs:
 
   profile:
     <<: *contrib_job
+    resource_class: large
```
Comment (jd) [inline .circleci/config.yml:417]: This looks unrelated?

**Review by jd — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (Kyle-Verhoog) [reply .circleci/config.yml:417]: yeah somewhat, just wanted my CI to be faster 😎

**Comment by Kyle-Verhoog** [comment]
```
collection failure
ImportError while importing test module '/root/project/tests/contrib/grpc_aio/test_grpc_aio.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../.pyenv/versions/3.10.3/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/contrib/grpc_aio/test_grpc_aio.py:19: in <module>
    from tests.contrib.grpc.hello_pb2 import HelloReply
tests/contrib/grpc/hello_pb2.py:10: in <module>
    from google.protobuf import descriptor as _descriptor
E   ModuleNotFoundError: No module named 'google.protobuf'
```

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/17865/workflows/4ae376e4-39e8-4910-a92a-29bb8a0e003b/jobs/1215507

🤔

**Review by jd — DISMISSED** [review]

**Review by jd — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by jd** [comment]
The grpc_aio` test fails because there is no protobuf installed. Not sure how it worked before?
The `grpc` job works because it pulls `googleapis-common-protos` which pulls protobuf. But I'm not even sure why the tests need `googleapis-common-protos` and not just `protobuf`?

**Review by jd — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 1.2
